### PR TITLE
Delete packages/flutter_tools/lib/src/fuchsia directory

### PR DESCRIPTION
**Pull request by zijiehe-google-com**
It's not being actively used, and fuchsia team does not have bandwidth to maintain it.

Bug: https://b.corp.google.com/issues/353729557

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Review by chingjun — APPROVED** [review]
Thanks for the cleanup. LGTM with a nit.

Also cc @christopherfujino @bkonyi

File: packages/flutter_tools/lib/src/flutter_application_package.dart [hunk 1]
```diff
@@ -104,9 +103,8 @@ class FlutterApplicationPackageFactory extends ApplicationPackageFactory {
             : WindowsApp.fromPrebuiltApp(applicationBinary);
       case TargetPlatform.fuchsia_arm64:
       case TargetPlatform.fuchsia_x64:
-        return applicationBinary == null
-            ? FuchsiaApp.fromFuchsiaProject(FlutterProject.current().fuchsia)
-            : FuchsiaApp.fromPrebuiltApp(applicationBinary);
+        // Unsupported yet.
+        return null;
```
[the remark this answers is not in the document]
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/flutter_application_package.dart:107]: This LGTM if you apply this suggestion

[the remark this answers is not in the document]
Comment (zijiehe-google-com) [reply packages/flutter_tools/lib/src/flutter_application_package.dart:107]: Done.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by zijiehe-google-com** [comment]
fuchsia does use flutter/engine, not very sure if the .ci.yaml file being referred above is relevant.

**Comment by zijiehe-google-com** [comment]
The pre built binaries from flutter/engine.

**Comment by zijiehe-google-com** [comment]
I will follow up with you offline.

**Pull request merged**